### PR TITLE
Use elastic-agent status as healthcheck

### PR DESCRIPTION
**Pull request by mtojek**
This PR modifies the healthcheck for elastic-agent and ~~fleet-server~~, so the correct one is used.

Blockers:
-  https://github.com/elastic/beats/issues/24956

### CODE REVIEW

**Comment by mtojek** [comment]
@ruflin We managed to switch to latest Docker image snapshots. Now it's the time to enable the right healthcheck. What is the healthcheck recommendation for the fleet-server?

**Comment by ruflin** [comment]
@michalpristas Instead of using `elastic-agent status`, could we also just start the agent monitoring, expose it and check there?

**Comment by michalpristas** [comment]
depends on the use case, if we want to wait for agent to be healthy and apps green, status is the right approach. also if agent is not able to connect to fleet it will return unhealthy status...
if we want to check for agent capability to run server we can go that way, but this wont reflect any failures or restarts of agent and its apps.

**Comment by ruflin** [comment]
I might have found a workaround here. If you add the following to the agent env variables it seems to work: 

```
- "STATE_PATH=/usr/share/elastic-agent"
```

It overwrites the default state path which is `/usr/share/elastic-agent/state`. I'm not sure about the other things this might break. The end solution should be that elastic-agent status respects the STATE_PATH set.

@mtojek Could you try if you get it running with the above env variable?

**Comment by mtojek** [comment]
@ruflin it seems to work, I'll retry the CI job to double-check. Do you think we should go with this workaround or is it better to wait for the exact fix? So far, the default Docker image is broken in terms of this feature.

File: internal/install/static_snapshot_yml.go [hunk 1]
```diff
@@ -102,14 +102,15 @@ services:
...
-      test: "sh -c 'grep \"Agent is starting\" -r . --include=elastic-agent-json.log'"
+      test: "elastic-agent status"
       retries: 90
       interval: 1s
     hostname: docker-fleet-agent
     environment:
     - "FLEET_ENROLL=1"
     - "FLEET_INSECURE=1"
     - "FLEET_URL=http://fleet-server:8220"
+    - "STATE_PATH=/usr/share/elastic-agent"
```
Comment (ruflin) [inline internal/install/static_snapshot_yml.go:113]: Should we change it for both fleet-server and elastic-agent?

**Review by ruflin — COMMENTED** [review]
I would like to get some feedback on @blakerouse on this before merging to make sure it does not have some unexpected side effects.

@mtojek It seems you just "run" the command, shouldn't there also be some check for the output itself?

**Comment by mtojek** [comment]
> @mtojek It seems you just "run" the command, shouldn't there also be some check for the output itself?

I expect the "exit code" to be the source of truth.

File: internal/install/static_snapshot_yml.go (hunk 1 above)
Comment (mtojek) [reply internal/install/static_snapshot_yml.go:113]: I can try and change it, but previously it didn't work well (maybe it was the STATE_PATH issue). I will re-check it :)

Comment (mtojek) [reply internal/install/static_snapshot_yml.go:114]: I updated the fleet-server to use "elastic-package status". Let's see.

Comment (blakerouse) [reply internal/install/static_snapshot_yml.go:114]: Why are you setting the `STATE_PATH`? The `/usr/share/elastic-agent/state` is the default, why not use that? Is this a workaround for the `status` command not working.

Comment (ruflin) [reply internal/install/static_snapshot_yml.go:114]: @blakerouse Yes, otherwise the status command does not work as it does it looks into the wrong path.

**Comment by mtojek** [comment]
I will re-test it once again to see if it works correctly.

@blakerouse would you mind sharing the recommendation for merging this change?

**Comment by mtojek** [comment]
Ok, something is unstable here: https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Felastic-package/detail/PR-329/9/pipeline/

I'm waiting for the timeout to grab other logs. Not sure if it's related to latest master or these changes.

**Comment by mtojek** [comment]
The error is not related to this PR. Something is wrong with the latest snapshot.

**Comment by mtojek** [comment]
It seems that the STATE_PATH did the trick for fleet-server as well!

**Comment by mtojek** [comment]
Nope, flaky again:

```
Attaching to elastic-package-stack_elastic-agent_1
[36melastic-agent_1              |[0m The Elastic Agent is currently in BETA and should not be used in production
[36melastic-agent_1              |[0m 
[36melastic-agent_1              |[0m Error: fail to enroll: fail to execute request to Kibana: could not decode the response, raw response: 
[36melastic-agent_1              |[0m 
[36melastic-agent_1              |[0m Error: enrollment failed: exit status 1
```

Should I switch to the previous healthcheck for the fleet-server?

**Comment by ruflin** [comment]
This does not really look related to the health status. I wonder if we hit a timeout? Can you reproduce it locally?

**Comment by mtojek** [comment]
I'm trying to reproduce it locally, but it always succeeds. Must be a really weird edge case. Call for improving logging around this place?

For reference:

```
untilfail ./k.sh
```

k.sh:
```
cat k.sh
#!/bin/bash

set -ex

elastic-package stack up -v -d
elastic-package stack down
```

**Comment by mtojek** [comment]
@ruflin @blakerouse any suggestions on how to solve this issue? Seems that the healthcheck is not ideal for the fleet server. Should I open a separate issue for this?

**Comment by ruflin** [comment]
@michalpristas What would you recommend for the health check for the fleet-server. The status command or the API endpoint?

@mtojek It seems now all checks have passed? Is it still flaky?

**Comment by mtojek** [comment]
> @mtojek It seems now all checks have passed? Is it still flaky?

Yes, I'm afraid so. With "api/status" it was the same problem, but I introduced a workaround - increased interval between next trials (higher chance to encounter the healthy state). I suspect it's related with restart of the agent.

**Review by ruflin — APPROVED** [review]
LGTM. I think we still need to work on "what is our recommended way" for docker.

**Pull request merged**